src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a standard contact parameter
+ */
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Name // instanceof handles nulls
```

Comment:
It is not correct to check whether _other_ is an instance of _Name_ class. It should be _Contact_ class since you have a common _equals_ methods in the parent class.